### PR TITLE
Fix metrics configuration in Sprint Boot application

### PR DESCRIPTION
**Pull request by bhatpmk**
# Description

This change sets the Spring Boot actuator endpoint /actuator/prometheus to the 'path' to scrape for metrics from the end point, in application configuration. New validations are added to assert the ES index and Prometheus scraped metrics.

Fixes VZ-2274

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/examples/springboot-app/springboot_test.go [hunk 1]
```diff
@@ -109,6 +112,63 @@ var _ = ginkgo.Describe("Verify Spring Boot Application", func() {
...
+				return appConfigMetricsExists()
+			}, longWaitTimeout, longPollingInterval).Should(gomega.BeTrue(), "Expected to find log index for Spring Boot application")
+		})
+	})
+
 })
+
+// appComponentMetricsExists checks whether component related metrics are available
+func appComponentMetricsExists() bool {
+	return pkg.MetricsExist("http_server_requests_seconds_count", "app_oam_dev_name", "springboot-appconf")
```
Comment (fredtibbitts) [inline tests/e2e/examples/springboot-app/springboot_test.go:147]: Minor nit, why create a function that just calls another function? Just inline this call above.

File: tests/e2e/examples/springboot-app/springboot_test.go [hunk 2]
```diff
@@ -109,6 +112,63 @@ var _ = ginkgo.Describe("Verify Spring Boot Application", func() {
...
+}
+
+// appConfigMetricsExists checks whether config metrics are available
+func appConfigMetricsExists() bool {
+	return pkg.MetricsExist("tomcat_sessions_created_sessions_total", "app_oam_dev_component", "springboot-component")
```
Comment (fredtibbitts) [inline tests/e2e/examples/springboot-app/springboot_test.go:152]: Same comment, just inline this.

File: tests/e2e/examples/springboot-app/springboot_test.go [hunk 3]
```diff
@@ -109,6 +112,63 @@ var _ = ginkgo.Describe("Verify Spring Boot Application", func() {
...
+				return logRecordFound(indexName)
+			}, longWaitTimeout, longPollingInterval).Should(gomega.BeTrue(), "Expected to find a recent log record")
+		})
+	})
+
+	ginkgo.Context("Verify Prometheus scraped metrics.", func() {
+		ginkgo.It("Retrieve Prometheus scraped metrics for App Component Metrics", func() {
+			gomega.Eventually(func() bool {
+				return appComponentMetricsExists()
+			}, longWaitTimeout, longPollingInterval).Should(gomega.BeTrue(), "Expected to find log index for Spring Boot application")
```
Comment (fredtibbitts) [inline tests/e2e/examples/springboot-app/springboot_test.go:134]: Error message text is wrong, refers to log index.

File: tests/e2e/examples/springboot-app/springboot_test.go [hunk 4]
```diff
@@ -109,6 +112,63 @@ var _ = ginkgo.Describe("Verify Spring Boot Application", func() {
...
+		})
+		ginkgo.It("Retrieve Prometheus scraped metrics for App Config Metrics", func() {
+			gomega.Eventually(func() bool {
+				return appConfigMetricsExists()
+			}, longWaitTimeout, longPollingInterval).Should(gomega.BeTrue(), "Expected to find log index for Spring Boot application")
```
Comment (fredtibbitts) [inline tests/e2e/examples/springboot-app/springboot_test.go:139]: Error message text is wrong, refers to log index.

**Review by fredtibbitts — CHANGES_REQUESTED** [review]

File: tests/e2e/examples/springboot-app/springboot_test.go (hunk 1 above)
Comment (bhatpmk) [reply tests/e2e/examples/springboot-app/springboot_test.go]: Done

File: tests/e2e/examples/springboot-app/springboot_test.go (hunk 3 above)
Comment (bhatpmk) [reply tests/e2e/examples/springboot-app/springboot_test.go]: Fixed.

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**